servers/rendering/renderer_rd/storage_rd/render_scene_buffers_rd.h

```diff
@@ -0,0 +1,273 @@
+#include "servers/rendering/renderer_rd/framebuffer_cache_rd.h"
+#include "servers/rendering/renderer_rd/storage_rd/render_scene_buffer_data_rd.h"
+#include "servers/rendering/renderer_scene.h"
+#include "servers/rendering/rendering_device.h"
+#include "servers/rendering/storage/render_scene_buffers.h"
+
+// These can be retired in due time
+#include "servers/rendering/renderer_rd/cluster_builder_rd.h"
+#include "servers/rendering/renderer_rd/effects/ss_effects.h"
+#include "servers/rendering/renderer_rd/environment/fog.h"
+
+class RenderSceneBuffersRD : public RenderSceneBuffers {
+	GDCLASS(RenderSceneBuffersRD, RenderSceneBuffers);
+
+private:
+	bool can_be_storage = true;
+	uint32_t max_cluster_elements = 512;
+	RD::DataFormat base_data_format = RD::DATA_FORMAT_R16G16B16A16_SFLOAT;
+	RendererRD::SSEffects *sse = nullptr;
+	RendererRD::VRS *vrs = nullptr;
+	uint64_t auto_exposure_version = 1;
+
+	// Our render target represents our final destination that we display on screen.
+	RID render_target;
+	Size2i target_size = Size2i(0, 0);
+	uint32_t view_count = 1;
+
+	// The internal size of the textures we render 3D to in case we render at a lower resolution and upscale
+	Size2i internal_size = Size2i(0, 0);
+	float fsr_sharpness = 0.2f;
+
+	// Aliassing settings
+	RS::ViewportMSAA msaa = RS::VIEWPORT_MSAA_DISABLED;
+	RS::ViewportScreenSpaceAA screen_space_aa = RS::VIEWPORT_SCREEN_SPACE_AA_DISABLED;
+	bool use_taa = false;
+	bool use_debanding = false;
+
+	// Named Textures
+
+	struct NTKey {
+		StringName context;
+		StringName buffer_name;
+
+		bool operator==(const NTKey &p_val) const {
+			return (context == p_val.context) && (buffer_name == p_val.buffer_name);
+		}
+
+		static uint32_t hash(const NTKey &p_val) {
+			// FIXME, properly hash two stringnames together
+			uint32_t h = p_val.context.hash();
+			h = hash_murmur3_one_32(p_val.buffer_name.hash(), h);
+			return hash_fmix32(h);
+		}
+
+		NTKey() {}
+		NTKey(const StringName p_context, const StringName p_texture_name) {
+			context = p_context;
+			buffer_name = p_texture_name;
+		}
+	};
+
+	struct NamedTexture {
+		// Cache the data used to create our texture
+		RD::TextureFormat format;
+		bool is_unique; // If marked as unique, we return it into our pool
+
+		// Our texture objects, slices are lazy (i.e. only created when requested).
+		RID texture;
+		Vector<RID> slices;
+	};
+
+	mutable HashMap<NTKey, NamedTexture, NTKey> named_textures;
+	void free_named_texture(NamedTexture &p_named_texture);
+
+	// Data buffers
+	mutable HashMap<StringName, Ref<RenderSceneBufferDataRD>> data_buffers;
+
+protected:
+	static void _bind_methods();
+
+public:
+	RenderSceneBuffersRD();
+	virtual ~RenderSceneBuffersRD();
+
+	// info from our renderer
+	void set_can_be_storage(const bool p_can_be_storage) { can_be_storage = p_can_be_storage; }
+	void set_max_cluster_elements(const uint32_t p_max_elements) { max_cluster_elements = p_max_elements; }
```

Comment:
Cluster should be move to a CustomData I guess too.

Reply:
Yup, but I think there is a fair amount of work left to extract the common logic from the cluster builder that shouldn't have been in the cluster builder to begin with, so that one I want to leave for a separate PR.